src/main/java/net/imglib2/img/cell/CellCursor.java

```diff
@@ -49,6 +50,8 @@
 {
 	protected final T type;
 
+	protected final Index i;
+
```

Comment:
Yes, I also didn't like that. I didn't want to make any changes unrelated to the problem for now.

We could swap the names 'i' and 'index' in the CellCursor.
Maybe the existing field could be removed, we would need to do benchmarks though. I remember that we spent a lot of time tweaking performance but that was before JMH. Maybe the current polymorphism issue played into the measurements and also maybe the JIT changed to favour different patterns.

In any case we will have to check that there is no derived class somewhere that uses the existing 'index'.

It would be good to revisit, but in a separate issue.
(Also #304 that I stumbled across while making the changes...)

Reply:
I found this [this class] in KNIME image processing that overloads CellCursor and uses the `index` field
https://github.com/knime-ip/knip/blob/a2e6aed508685a35d43c964188ef6627f6206683/org.knime.knip.core/src/org/knime/knip/core/io/externalization/externalizers/CellImgExt1.java#L183

Reply:
Let's either merge this branch as it is, or rename all the fields to something consistent, like `typeIndex`.

Reply:
I find `typeIndex` a good compromise as it is unambiguous and is unlikely to create surprises (such as in `for (int i ...` loops).